lib/postcode_validation/use_case/validate_address.rb

```diff
@@ -35,6 +36,28 @@ def potential_address_matches
         @address_match_gateway.query(search_term: postcode,
                                      country: country)
       end
+
+      def invalid_postcode_format?
+        downcase_country = country.downcase
+        regex = send("#{downcase_country}_postcode_regex")
```

Comment:
I'd refactor this into another class so that it's polymorphic (some psuedocode)

Factory pattern is commonly used to move the construction code out...

```ruby
validator = format_validator_factory.validator(country: downcase_country)
validator.valid?(postcode)
```

```ruby
class FormatValidatorFactory
   def validator(country:)
     case country
       when "nl" NLPostcodeValidator.new
       when "gb" GBPostcodeValidator.new
       else AlwaysValidPostcodeValidator.new #maybe?
     end
   end
end
```

```ruby
class NLPostcodeValidator
    def valid?(postcode)
      /(NL-)?(\d{4})\s*([A-Z]{2})/.match?(postcode)
    end
end
```